src/Microsoft.Health.Dicom.Api/Features/ByteCounter/IResponseLogStreamFactory.cs

```diff
@@ -0,0 +1,19 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.IO;
+
+namespace Microsoft.Health.Dicom.Api.Features.ByteCounter
+{
+    public interface IResponseLogStreamFactory
+    {
+        /// <summary>
+        /// Creates a <see cref="ByteCountingStream"/>
+        /// </summary>
+        /// <param name="stream">An underlying stream.</param>
+        /// <returns>A byte counting stream that wraps the specified underlying stream.</returns>
+        ByteCountingStream CreateByteCountingResponseLogStream(Stream stream);
```

Comment:
Any reason we're deviating from the factory pattern we have elsewhere in the code base? [IGuidFactory](https://github.com/microsoft/dicom-server/blob/main/src/Microsoft.Health.Dicom.Functions/Durable/IGuidFactory.cs) or [IMultiPartReadFactory](https://github.com/microsoft/dicom-server/blob/main/src/Microsoft.Health.Dicom.Core/Web/IMultipartReaderFactory.cs)? It seems like this would be called a `IByteCountingStreamFactory` with a method `IByteCountingStream Create(Stream stream)`

Reply:
I was just keeping the same names used for fhir but this makes sense. I made the rename changes in this PR if you could take a look: https://github.com/microsoft/healthcare-shared-components/pull/314

I will delete the ByteCounter/ files from this PR once that is merged in and I can reference those files from here

Reply:
Code has been checked in to healthcare-shared-components and deleted here.